lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c

```diff
@@ -1212,21 +1372,42 @@ int s1ap_mme_handle_ue_context_release_complete(
 int s1ap_mme_handle_initial_context_setup_failure(
     s1ap_state_t* state, __attribute__((unused)) const sctp_assoc_id_t assoc_id,
     __attribute__((unused)) const sctp_stream_id_t stream,
-    struct s1ap_message_s* message) {
-  S1ap_InitialContextSetupFailureIEs_t* initialContextSetupFailureIEs_p = NULL;
-  ue_description_t* ue_ref_p                                            = NULL;
-  MessageDef* message_p                                                 = NULL;
+    S1ap_S1AP_PDU_t* pdu) {
+  S1ap_InitialContextSetupFailure_t* container;
+  S1ap_InitialContextSetupFailureIEs_t* ie = NULL;
+  ue_description_t* ue_ref_p               = NULL;
+  MessageDef* message_p                    = NULL;
   S1ap_Cause_PR cause_type;
   long cause_value;
-  int rc          = RETURNok;
-  imsi64_t imsi64 = INVALID_IMSI64;
+  int rc                          = RETURNok;
+  imsi64_t imsi64                 = INVALID_IMSI64;
+  mme_ue_s1ap_id_t mme_ue_s1ap_id = 0;
+  enb_ue_s1ap_id_t enb_ue_s1ap_id = 0;
 
   OAILOG_FUNC_IN(LOG_S1AP);
-  initialContextSetupFailureIEs_p =
-      &message->msg.s1ap_InitialContextSetupFailureIEs;
+  container =
+      &pdu->choice.unsuccessfulOutcome.value.choice.InitialContextSetupFailure;
+
+  S1AP_FIND_PROTOCOLIE_BY_ID(
+      S1ap_InitialContextSetupFailureIEs_t, ie, container,
+      S1ap_ProtocolIE_ID_id_MME_UE_S1AP_ID, true);
+  if (ie) {
```

Comment:
The braces can be simplified by reversing the if condition
```suggestion
  if (!ie) {
     OAILOG_FUNC_RETURN(LOG_S1AP, RETURNok);
  }
  mme_ue_s1ap_id = ie->value.choice.MME_UE_S1AP_ID;   
```

Reply:
Resolved